include/nbl/asset/filters/kernels/WeightFunctions.h

```diff
@@ -0,0 +1,308 @@
+#ifndef _NBL_ASSET_MATH_FUNCTIONS_H_INCLUDED_
+#define _NBL_ASSET_MATH_FUNCTIONS_H_INCLUDED_
+
+#include <limits>
+#include <cstdint>
+
+namespace nbl::asset
+{
+
+struct SDiracFunction
+{
+	// We can use std::nextafter after C++23
+	constexpr static inline float min_support = -1e-6f;
+	constexpr static inline float max_support = +1e-6f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template<int32_t derivative=0>
+	static inline double weight(float x)
+	{
+		if (x != 0.f)
+			return 0.0;
+
+		if constexpr (derivative == 0)
+		{
+			return std::numeric_limits<double>::infinity();
+		}
+		else
+		{
+			static_assert(false);
+			return core::nan<double>();
+		}
+	}
+};
+
+// Standard Box function, symmetric, value in the support is 1, integral is 1, so support must be [-1/2,1/2)
+struct SBoxFunction
+{
+	constexpr static inline float min_support = -0.5f;
+	constexpr static inline float max_support = +0.5f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template <int32_t derivative = 0>
+	static inline double weight(float x)
+	{
+		if (x >= min_support && x < max_support)
+		{
+			if constexpr (derivative == 0)
+			{
+				return 1.0;
+			}
+			else
+			{
+				static_assert(false);
+				return core::nan<double>();
+			}
+		}
+		return 0.0;
+	}
+};
+
+// Standard Triangle function, symmetric, peak in the support is 1 and at origin, integral is 1, so support must be [-1,1)
+struct STriangleFunction
+{
+	constexpr static inline float min_support = -1.f;
+	constexpr static inline float max_support = +1.f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template <int32_t derivative = 0>
+	static inline double weight(float x)
+	{
+		if (x >= min_support && x < max_support)
+		{
+			if constexpr (derivative > 0)
+			{
+				// Derivative at 0 not defined.
+				static_assert(false);
+				return core::nan<double>();
+			}
+			else
+			{
+				return 1.0 - core::abs(x);
+			}
+		}
+		return 0.0;
+	}
+};
+
+// Truncated Gaussian function, with stddev = 1.0, if you want a different stddev then you need to scale it.
+struct SGaussianFunction
+{
+	constexpr static inline float min_support = -3.f;
+	constexpr static inline float max_support = +3.f;
+	constexpr static inline uint32_t k_smoothness = std::numeric_limits<uint32_t>::max();
+
+	template <int32_t derivative = 0>
+	static inline double weight(float x)
+	{
+		if (x >= min_support && x < max_support)
+		{
+			if constexpr (derivative == 0)
+			{
+				const double normalizationFactor = core::inversesqrt(2.0 * core::PI<double>()) / std::erff(core::sqrt<double>(2.0) * double(min_support));
```

Comment:
the denominator is wrong.

Its supposed to be `std::erff(core::inversesqrt(2.0)*double(max_support))`

because CDF (which also is the integral from -INF to x) is `0.5+0.5*erf(x/sqrt(2))`

so to get definite integral, you have `0.5(erf(x/sqrt(2))-erf(-x/sqrt(2)))` which due to erf identity becomes `erf(x/sqrt(2))`

Reply:
I see you copied my original code, seems it had a bug with the squareroot inside the erf not being a reciprocal :(

Reply:
fixed it myself.